### ISSUE TITLE
Manipulations with Key

### ISSUE THREAD

**Issue by AuHau** [ipfs/interface-datastore#30]
If my understanding is correct, `Key` is immutable. I believe there should be set of functions that will allow to manipulate with `Key`s. For example `join()` operation would be handy, because you don't export separator, so I assume you just hardcode `/` if it is needed, which does not seem like a good solution to me.

What operations would make sense? I thought of these:

 - `join(key1, key2)` => `new Key(key1 + key2)`
 - `namespaceMap(new Key('SOME/key'), (namespace) => namespace.lower())` ==> `Key('some/key')`

Allow of these actions should return new instance with the modified values.

How this should be implemented? I would say as a static method of the `Key` class?

**Comment by achingbrain** [comment]
I like it.  Instead of static methods, how about following the js `Array` convention of `Array.prototype.concat([value1[, value2[, ...[, valueN]]]])` when mutating existing keys?

So:

```javascript
const key1 = new Key('foo')
const key2 = key1.concat('bar', 'baz)

console.info(key2)
// '/foo/bar/baz'
```

**Comment by AuHau** [comment]
Sweet! Thanks for the input!

Yeah, that was the second solution that I had in mind. To be honest, I don't have any strong preference, so I am alright with the `Array` approach.

What about the set of operations? Do you have some other ideas what could be included?

**Comment by achingbrain** [comment]
> Do you have some other ideas what could be included?

I'm happy to take the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) approach - just add what we need, when we need it instead of trying to come up with everything ahead of time.

**Comment by AuHau** [comment]
Solved with #31, so closing this. If more operations would be needed, they can be integrated separately.
